src/computational_graph/graph.jl

```diff
@@ -400,36 +400,50 @@ function standardize_order!(g::Graph)
     end
 end
 
-prune_unary(g::Graph) = ((length(g.subgraph) == 1 && g.subgraph_factors[1] == 1 && g.factor == 1) ? g.subgraph[1] : g)
+function prune_unary(g::Graph)
+    if (g.operator in [Prod, Sum] && length(g.subgraph)==1 && g.subgraph_factors[1]==1 && g.factor == 1)
```

Comment:
g.subgraph_factors[1] \approx one(F) && g.factor \approx one(F).
use \approx to compare non-integer values, one(F) is the identity element of the generic type F